quality-assurance/core.md

```diff
@@ -0,0 +1,192 @@
+# 🚦 Langley Foxall Quality Assurance
+
+When building applications that run an entire business end-to-end - it's important that we adhere to high quality
+assurance standards in order to consistently deliver a high quality software project.
+
+There are different parts/processes to quality assurance as a whole, and these are not limited to just manual testing and
+automated testing.
+
+*   [What makes a good web application?](#what-makes-a-good-web-application)
+    *   [Blackbox](#blackbox)  
+    *   [Whitebox](#whitebox)  
+*   [Processes](#processes)
+    *   [Automated Testing](#automated-testing)
+    *   [Manual Testing](#manual-testing)
+
+## What makes a good web application?
+The following is a checklist of things to think about both during development of a product and during the quality 
+assurance phase. They should be considered guidelines, and not a strict set of requirements, as not all points will be applicable or appropriate for all applications.
+
+### Blackbox
+**General**
+* All UI scales well and looks good on small form factor devices
+* All errors are handled and UI reacts appropriately
+* Errors are clear in meaning &amp; reflect what happened
+* Activity indicators are used when loading data asynchronously
+* Empty information is handled correctly in UI (optional fields etc)
+* Long running operations give a indication of progress if possible
+* Breadcrumbs are used where appropriate
+
+**Navbar**
+* UI reflects the current active page
+* An icon that is appropriate for the page (if applicable)
+
+**Forms**
+* Invalid inputs correctly trigger validation errors
+* All form inputs have associated labels
+* Appropriate validation for different inputs (phone number, postcode)
+* The forms are not too long &amp; and are grouped logically
+* The user is alerted of the form lifecycle (success, error)
+* When modals are used in the context of updates, ensure inputs are correctly pre populated.
+* Autofill is disabled on inputs where appropriate
+* Inputs have correct type property (emails have email type, phone numbers have tel type)
+
+**Modals**
+* Modals are animated on both display and dismiss
+* Modals are used where appropriate over pages
+
+**Formatting**
+* Dates are formatted correctly for the appropriate audience.
+* Monetary values are formatted correctly for the appropriate audience
+* Phone numbers are formatted correctly for the appropriate audience
+
+**Pagination**
+* UI correctly indicates current page
+* UI shows at least 2 page tabs ahead and below the current page
+
+**Login &amp; Auth**
+* Users can only access pages they are permitted access to.
+* Authentication when they are logged in
+* Login, registration (if applicable) and logout works as expected.
+* Reset password works as expected
+
+**Registration**
+* Email addresses are verified &amp; unique (if applicable)
+* Passwords meet minimum standards (min 8 chars)
```

Comment:
Should we enforce usage of https://github.com/langleyfoxall/laravel-nist-password-rules?

Reply:
We could do something similar to the following.

```suggestion
* Passwords meet minimum standards, as defined by [NIST Special Publication 800-63B section 5](https://pages.nist.gov/800-63-3/sp800-63b.html#sec5). If using Laravel, ensure the [Laravel NIST Password Rules](https://github.com/langleyfoxall/laravel-nist-password-rules) is integrate.
```

Reply:
I'm against blanket enforcing the NIST rules due to their complexities and it would cause a significant barrier to entry for social apps (registration processes) and uptake to those platforms.

For that reason I think a lower level as a minimum (8 Chars) is enough, but its _preferable_ in our enterprise software to opt for NIST specification.

Reply:
```suggestion
* For enterprise software, passwords should preferably meet standards defined by [NIST Special Publication 800-63B section 5](https://pages.nist.gov/800-63-3/sp800-63b.html#sec5). If using Laravel, ensure the [Laravel NIST Password Rules](https://github.com/langleyfoxall/laravel-nist-password-rules) is integrated.
```

@wheatleyjj New suggestion based on your feedback.